### ISSUE TITLE
Requirements unable to be installed

### ISSUE THREAD

**Issue by Proasek** [zjp-shadow/CharacterGen#26]
Trying to follow the install instructions, but I run into an error when running "pip install -r requirements.txt", reading:
"ERROR: Could not find a version that satisfies the requirement torch (from versions: none) 
ERROR: No matching distribution found for torch"

I don't work in command line if I can help it so I'm not really sure where to go from here, I've tried installing pytorch separately but that did not fix it. I'm using python 3.9.0 as instructed on Windows 11.

**Comment by Lucalme** [comment]
To install pytorch separately visit this page and copy the custom command into your cmd or powershell:
https://pytorch.org/get-started/locally/

ie if you have an Nvidia GPU with the CUDA toolkit version 12.4 installed on your system, your command should look like this:
>pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124

once pytorch is installed correctly, just edit the requirements.txt and put # in front of  torch and torchvision like this:
<img width="637" alt="image" src="https://github.com/user-attachments/assets/958b109d-71c7-45d3-8aff-a4c715a94839">
and save it.


then re-run the command:
>pip install -r requirements.txt

and you should be good to go

**Comment by Proasek** [comment]
> To install pytorch separately visit this page and copy the custom command into your cmd or powershell: https://pytorch.org/get-started/locally/
> 
> ie if you have an Nvidia GPU with the CUDA toolkit version 12.4 installed on your system, your command should look like this:
> 
> > pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124
> 
> once pytorch is installed correctly, just edit the requirements.txt and put # in front of torch and torchvision like this: <img alt="image" width="637" src="https://private-user-images.githubusercontent.com/46386244/360538550-958b109d-71c7-45d3-8aff-a4c715a94839.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzOTUyNjMsIm5iZiI6MTcyNDM5NDk2MywicGF0aCI6Ii80NjM4NjI0NC8zNjA1Mzg1NTAtOTU4YjEwOWQtNzFjNy00NWQzLThhZmYtYTRjNzE1YTk0ODM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDA2MzYwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzMTQ0ODM1OWY5N2RmMTc1ZjdlY2NlMmQwMGE4YTg2NzI0NjZkZmFkZDliNmVjY2QyYzg4NmIzZjQ2ZGU4ZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O85ycsmIfkPghJEjnT30qIvIiMsq4utAUEOwckMvfcM"> and save it.
> 
> then re-run the command:
> 
> > pip install -r requirements.txt
> 
> and you should be good to go
> 
> edit: If you have a Nvidia GPU but haven't installed the CUDA toolkit, you must follow a guide on how to set it up properly for your system. I recommend installing a version listed on https://pytorch.org/get-started/locally/ (preferably 12.4) so you will have less trouble with compatibility issues.

The same error occurs when trying to install torch separately from that page's suggested commands.

**Comment by Lucalme** [comment]
> > To install pytorch separately visit this page and copy the custom command into your cmd or powershell: https://pytorch.org/get-started/locally/
> > ie if you have an Nvidia GPU with the CUDA toolkit version 12.4 installed on your system, your command should look like this:
> > > pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124
> > 
> > 
> > once pytorch is installed correctly, just edit the requirements.txt and put # in front of torch and torchvision like this: <img alt="image" width="637" src="https://private-user-images.githubusercontent.com/46386244/360538550-958b109d-71c7-45d3-8aff-a4c715a94839.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzOTUyNjMsIm5iZiI6MTcyNDM5NDk2MywicGF0aCI6Ii80NjM4NjI0NC8zNjA1Mzg1NTAtOTU4YjEwOWQtNzFjNy00NWQzLThhZmYtYTRjNzE1YTk0ODM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDA2MzYwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzMTQ0ODM1OWY5N2RmMTc1ZjdlY2NlMmQwMGE4YTg2NzI0NjZkZmFkZDliNmVjY2QyYzg4NmIzZjQ2ZGU4ZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O85ycsmIfkPghJEjnT30qIvIiMsq4utAUEOwckMvfcM"> and save it.
> > then re-run the command:
> > > pip install -r requirements.txt
> > 
> > 
> > and you should be good to go
> > edit: If you have a Nvidia GPU but haven't installed the CUDA toolkit, you must follow a guide on how to set it up properly for your system. I recommend installing a version listed on https://pytorch.org/get-started/locally/ (preferably 12.4) so you will have less trouble with compatibility issues.
> 
> The same error occurs when trying to install torch separately from that page's suggested commands.


What happens when you try to install other modules (ie : `pip install einops` ) ?
You may want to upgrade your pip version : `python -m install pip==24.2` or `python -m pip install -U pip`

**Comment by Proasek** [comment]
> > > To install pytorch separately visit this page and copy the custom command into your cmd or powershell: https://pytorch.org/get-started/locally/
> > > ie if you have an Nvidia GPU with the CUDA toolkit version 12.4 installed on your system, your command should look like this:
> > > > pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124
> > > 
> > > 
> > > once pytorch is installed correctly, just edit the requirements.txt and put # in front of torch and torchvision like this: <img alt="image" width="637" src="https://private-user-images.githubusercontent.com/46386244/360538550-958b109d-71c7-45d3-8aff-a4c715a94839.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzOTUyNjMsIm5iZiI6MTcyNDM5NDk2MywicGF0aCI6Ii80NjM4NjI0NC8zNjA1Mzg1NTAtOTU4YjEwOWQtNzFjNy00NWQzLThhZmYtYTRjNzE1YTk0ODM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDA2MzYwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzMTQ0ODM1OWY5N2RmMTc1ZjdlY2NlMmQwMGE4YTg2NzI0NjZkZmFkZDliNmVjY2QyYzg4NmIzZjQ2ZGU4ZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O85ycsmIfkPghJEjnT30qIvIiMsq4utAUEOwckMvfcM"> and save it.
> > > then re-run the command:
> > > > pip install -r requirements.txt
> > > 
> > > 
> > > and you should be good to go
> > > edit: If you have a Nvidia GPU but haven't installed the CUDA toolkit, you must follow a guide on how to set it up properly for your system. I recommend installing a version listed on https://pytorch.org/get-started/locally/ (preferably 12.4) so you will have less trouble with compatibility issues.
> > 
> > 
> > The same error occurs when trying to install torch separately from that page's suggested commands.
> 
> What happens when you try to install other modules (ie : `pip install einops` ) ? You may want to upgrade your pip version : `python -m install pip==24.2` or `python -m pip install -U pip`

Einops installed without issue, and I'm already using pip 24.2. problem with torch persists.

**Comment by Lucalme** [comment]
> > > > To install pytorch separately visit this page and copy the custom command into your cmd or powershell: https://pytorch.org/get-started/locally/
> > > > ie if you have an Nvidia GPU with the CUDA toolkit version 12.4 installed on your system, your command should look like this:
> > > > > pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124
> > > > 
> > > > 
> > > > once pytorch is installed correctly, just edit the requirements.txt and put # in front of torch and torchvision like this: <img alt="image" width="637" src="https://private-user-images.githubusercontent.com/46386244/360538550-958b109d-71c7-45d3-8aff-a4c715a94839.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzOTUyNjMsIm5iZiI6MTcyNDM5NDk2MywicGF0aCI6Ii80NjM4NjI0NC8zNjA1Mzg1NTAtOTU4YjEwOWQtNzFjNy00NWQzLThhZmYtYTRjNzE1YTk0ODM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDA2MzYwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzMTQ0ODM1OWY5N2RmMTc1ZjdlY2NlMmQwMGE4YTg2NzI0NjZkZmFkZDliNmVjY2QyYzg4NmIzZjQ2ZGU4ZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O85ycsmIfkPghJEjnT30qIvIiMsq4utAUEOwckMvfcM"> and save it.
> > > > then re-run the command:
> > > > > pip install -r requirements.txt
> > > > 
> > > > 
> > > > and you should be good to go
> > > > edit: If you have a Nvidia GPU but haven't installed the CUDA toolkit, you must follow a guide on how to set it up properly for your system. I recommend installing a version listed on https://pytorch.org/get-started/locally/ (preferably 12.4) so you will have less trouble with compatibility issues.
> > > 
> > > 
> > > The same error occurs when trying to install torch separately from that page's suggested commands.
> > 
> > 
> > What happens when you try to install other modules (ie : `pip install einops` ) ? You may want to upgrade your pip version : `python -m install pip==24.2` or `python -m pip install -U pip`
> 
> Einops installed without issue, and I'm already using pip 24.2. problem with torch persists.

I just came across a forum post of a person who has had the same issue you have, the answer mentionned assuring you have a 64 bits version of python and not a 32 bits. You can find the post and the answer [here](https://stackoverflow.com/a/76840428)

tldr: the `python -c "import sys; print(sys.maxsize > 2**32)"` command should return True when you run it in your cmd. 

Let me know if that does it or if the problem persists.

**Comment by Proasek** [comment]
> > > > > To install pytorch separately visit this page and copy the custom command into your cmd or powershell: https://pytorch.org/get-started/locally/
> > > > > ie if you have an Nvidia GPU with the CUDA toolkit version 12.4 installed on your system, your command should look like this:
> > > > > > pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124
> > > > > 
> > > > > 
> > > > > once pytorch is installed correctly, just edit the requirements.txt and put # in front of torch and torchvision like this: <img alt="image" width="637" src="https://private-user-images.githubusercontent.com/46386244/360538550-958b109d-71c7-45d3-8aff-a4c715a94839.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzOTUyNjMsIm5iZiI6MTcyNDM5NDk2MywicGF0aCI6Ii80NjM4NjI0NC8zNjA1Mzg1NTAtOTU4YjEwOWQtNzFjNy00NWQzLThhZmYtYTRjNzE1YTk0ODM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDA2MzYwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzMTQ0ODM1OWY5N2RmMTc1ZjdlY2NlMmQwMGE4YTg2NzI0NjZkZmFkZDliNmVjY2QyYzg4NmIzZjQ2ZGU4ZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O85ycsmIfkPghJEjnT30qIvIiMsq4utAUEOwckMvfcM"> and save it.
> > > > > then re-run the command:
> > > > > > pip install -r requirements.txt
> > > > > 
> > > > > 
> > > > > and you should be good to go
> > > > > edit: If you have a Nvidia GPU but haven't installed the CUDA toolkit, you must follow a guide on how to set it up properly for your system. I recommend installing a version listed on https://pytorch.org/get-started/locally/ (preferably 12.4) so you will have less trouble with compatibility issues.
> > > > 
> > > > 
> > > > The same error occurs when trying to install torch separately from that page's suggested commands.
> > > 
> > > 
> > > What happens when you try to install other modules (ie : `pip install einops` ) ? You may want to upgrade your pip version : `python -m install pip==24.2` or `python -m pip install -U pip`
> > 
> > 
> > Einops installed without issue, and I'm already using pip 24.2. problem with torch persists.
> 
> I just came across a forum post of a person who has had the same issue you have, the answer mentionned assuring you have a 64 bits version of python and not a 32 bits. You can find the post and the answer [here](https://stackoverflow.com/a/76840428)
> 
> tldr: the `python -c "import sys; print(sys.maxsize > 2**32)"` command should return `True` when you run it in your cmd. If it does not you should consider reinstalling python.
> 
> Let me know if that does it or if the problem persists.

This did it! Turns out at some point I'd managed to install both the 32 bit AND the 64 bit versions, and the former was taking priority. Thank you so much for your help!

**Comment by Proasek** [comment]
Issue resolved, cause: using 32 bit version of python instead of 64.
